### PR TITLE
chore(build): add external api for creating derived extensions

### PR DESCRIPTION
**Pull request by jrbotros**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,17 @@
 {
+  "name": "jrbotros-test-tent",
+  "version": "0.1.24",
+  "description": "Tent is an extension for Google Chrome allowing users to map web content to a list of annotations.",
+  "author": "Joseph Botros <me@jrbotros.com> (http://jrbotros.com/)",
```
Comment (marcua) [inline package.json]: Team B12 or B12?

https://github.com/b12io/orchestra/blob/master/package.json#L56
https://github.com/segmentio/nightmare/blob/master/package.json#L13

File: Tent.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+const assignIn = require('lodash.assignin');
```
Comment (marcua) [inline Tent.js]: add usage info to README

File: Tent.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+const pick = require('lodash.pick');
+const semver = require('semver');
+const shell = require('shelljs');
+const tmp = require('tmp');
+
+const manifestJSON = require('./manifest.json');
+const packageJSON = require('./package.json');
+
+shell.set('-e');
+
```
Comment (marcua) [inline Tent.js]: document args like apiFile, bumpVerson, manifestOverrides, outputPath

File: package.json [hunk 4]
```diff
@@ -1,7 +1,17 @@
...
   "scripts": {
     "package": "bash scripts/package.sh"
   },
+  "main": "Tent.js",
+  "license": "MIT",
```
Comment (marcua) [inline package.json]: Apache License 2.0

File: package.json [hunk 5]
```diff
@@ -1,7 +1,17 @@
 {
+  "name": "jrbotros-test-tent",
+  "version": "0.1.24",
```
Comment (jrbotros) [inline package.json]: Restore this to `0.1.0` and publish to correct name

File: package.json (hunk 1 above)
Comment (jrbotros) [inline package.json]: I like `B12`!

File: package.json (hunk 4 above)
Comment (jrbotros) [inline package.json]: Why?

File: Tent.js (hunk 3 above)
Comment (jrbotros) [inline Tent.js]: I'm going to document the external API in a separate PR once #38 is landed, let's get this in first!

File: package.json (hunk 1 above)
Comment (jrbotros) [inline package.json]: ✅

File: package.json (hunk 4 above)
Comment (jrbotros) [inline package.json]: (✅)

File: LICENSE [hunk 6]
```diff
@@ -1,21 +1,201 @@
...
+To apply the Apache License to your work, attach the following
+boilerplate notice, with the fields enclosed by brackets "{}"
+replaced with your own identifying information. (Don't include
+the brackets!)  The text should be enclosed in the appropriate
+comment syntax for the file format. We also recommend that a
+file or class name and description of purpose be included on the
+same "printed page" as the copyright notice for easier
+identification within third-party archives.
+
+Copyright {yyyy} {name of copyright owner}
```
Comment (marcua) [inline LICENSE]: 2017 B12

**Pull request merged**